### ISSUE TITLE
ProfileQueryEndpoint is not overridable => application identifiers cannot be resolved

### ISSUE THREAD

**Issue by zakimaksyutov** [Microsoft/ApplicationInsights-dotnet-server#853]
## Description
Application ID resolution feature does not work in our own PPE environment. As a result we cannot use new correlation-based features for monitoring our production.

## Repro Steps
1. Override EndpointAddress to PPE endpoint

##  Actual Behavior
Notice that application identifiers are still resolved (ProfileQueryEndpoint) against original endpoint. Application Insights doesn't allow to configure this endpoint. 

## Expected Behavior
Allow applications to configure ProfileQueryEndpoint or construct based on EndpointAddress

## Version Info 
SDK Version : 2.6-beta2

## Code

```
using System.Net.Http;
using System.Threading;
using Microsoft.ApplicationInsights;
using Microsoft.ApplicationInsights.DataContracts;
using Microsoft.ApplicationInsights.Extensibility;
using Microsoft.ApplicationInsights.Extensibility.Implementation;

namespace OverrideProfileApp
{
    class Program
    {
        public static readonly HttpClient client = new HttpClient();

        static void Main(string[] args)
        {
            TelemetryConfiguration.Active.InstrumentationKey = "11111111-1111-1111-1111-111111111111";
            TelemetryConfiguration.Active.TelemetryChannel.EndpointAddress =
                "https://myingestionpoint.com/v2/track";

            var a = TelemetryModules.Instance;

            var telemetryClient = new TelemetryClient();

            while (true)
            {
                Thread.Sleep(1000);

                using (var operation = telemetryClient.StartOperation<RequestTelemetry>("MyOperation"))
                {
                    client.GetStringAsync("https://bing.com").Wait();
                }
            }
        }
    }
}
```

**Comment by SergeyKanzhelev** [comment]
This is by design. Please configure `ProfileQueryUrl` explicitly instead of taking unnecessary dependency on channel's `EndpointAddress`. Two reasons - 1. `EndpointAddress` property may throw `NotImplementedException` for the non-standard channels 2. If proxy is specified for telemetry upload - this proxy is not necessarily proxies profile requests.

**Comment by zakimaksyutov** [comment]
Chatted with @SergeyKanzhelev - the issue of not being able to override `ProfileQueryUrl` is still an issue =) Agree that it should be a different setting and not depend on `EndpointAddress`. Reactivating.

**Comment by SergeyKanzhelev** [comment]
sorry for miunderstanding. Definitely should be fixed

**Comment by MS-TimothyMothra** [comment]
Adding some context to the conversation while planning how to fix this...

This affects the `CorrelationIdLookupHelper`'s `endpointAddress` which is responsible for resolving the Application Id.
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/113cf9ec4aa3dfd0d1c6443ded1700a0fa65972b/Src/Common/CorrelationIdLookupHelper.cs#L82


Our current documentation says to override the TelemetryChannel's EndpointAddress if a user is behind a proxy. This is questionably incorrect. [Microsoft Docs: "AppInsights Proxy"](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-troubleshoot-faq#proxy)


Currently today, `ProfileQueryEndpoint' is copied to dependent modules. As a workaround, It should be possible to overwrite these manually, although that's a far from ideal scenario for an end-user.

https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/33696c0d09a3c6e93a690198a09bd49d84a63c58/Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs#L83-L94

https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/e95d9caac611d3da55879fefba149f4d793dffff/Src/DependencyCollector/Shared/DependencyTrackingTelemetryModule.cs#L98-L109

**Comment by MS-TimothyMothra** [comment]
@zakimaksyutov I've got a workaround for you until we release a more permanent solution.

Run this snippet after customizing your config:

```
foreach (var module in TelemetryModules.Instance.Modules)
{
    module.Initialize(TelemetryConfiguration.Active);
}
```
This will ensure that ANY config customizations are applied to all modules.

**Comment by zakimaksyutov** [comment]
@MS-TimothyMothra, tried to add above snippet to above repro and it doesn't look like that it helped. AppIDs are still resolved against main endpoint.

**Comment by MS-TimothyMothra** [comment]
Fix will be released with 2.6.0-Beta4
